lib/snowplow-tracker/tracker.rb

```diff
@@ -122,38 +120,58 @@ def build_context(context)
 
     # Tracking methods
 
-    # Attaches all the fields in @standard_nv_pairs to the request
+    # Attaches all the fields in @settings to the request
     # Only attaches the context vendor if the event has a custom context
     #
     Contract Payload => nil
     def track(payload)
-      payload.add_hash(@subject.standard_nv_pairs)
-      payload.add_hash(@standard_nv_pairs)
-      payload.add('eid', event_id)
       @emitters.each { |emitter| emitter.input(payload.context) }
 
       nil
     end
 
+    Contract Or[Timestamp, Num, nil] => Timestamp
+    def process_tstamp(tstamp)
+      tstamp = Timestamp.create if tstamp.nil?
+      tstamp = DeviceTimestamp.new(tstamp) if tstamp.is_a? Numeric
+      tstamp
+    end
+
```

Comment:
I'm just looking at this and wondering if it's historically incorrect. With the introduction of `true_timestamp` we give users two timestamps to play with. Ideally users shouldn't really set the device timestamp (`dtm`) but we continue letting them. However, if the true timestamp is set, then the `dtm` property should still be set. If I'm reading this correctly, if the user passes a `ttm` (TrueTimestamp) into the `tstamp` property, then we won't set the `dtm` field. I think we could update `finalise_payload` to still set `dtm` (to current time) if `ttm` is set.